tests/PHPUnit/Integration/DataAccess/LogAggregatorTest.php

```diff
@@ -341,6 +341,71 @@ public function test_getSegmentTmpTableNameWithLongPrefix()
         $this->assertEquals('logtmpsegmentcc2efa0acbd5f209', $this->logAggregator->getSegmentTmpTableName());
```

Comment:
Could maybe be worth to add a test that uses the new join feature as well as max execution time hint.

Reply:
I've moved the query generation code from `queryVisitsByDimension` to a new sub-method `getQueryByDimensionSql` to make it easier to test the generated sql. I've then added a new test to check the expected visits by dimension query with an origin hint, join prefix hint and a max execution time hint.